lib/vfscore/automount.c

```diff
@@ -67,10 +71,103 @@ struct vfscore_volume {
 	unsigned long flags;
 	/* Mount options */
 	const char *opts;
+#if CONFIG_LIBVFSCORE_FSTAB
+	/* Unikraft Mount options, see vfscore_mount_volume() */
+	char *ukopts;
+#endif /* CONFIG_LIBVFSCORE_FSTAB */
 };
 
+#if CONFIG_LIBVFSCORE_FSTAB
+/* Handle `mkpath` Unikraft Mount Option */
+static int vfscore_volume_process_ukopts_do_mkpath(const char *path)
+{
+	const char *curri;
+	char *pos;
+	int rc;
+
+	UK_ASSERT(path);
+
+	curri = path;
+	pos = strchr(path + 1, '/');
+
+	while (pos) {
+		/* If we reached last directory, delete trailing `/` and stop */
+		if (pos[1] == '\0') {
```

Comment:
I guess we also need to catch `pos[0] == '\0'` before we look into `pos[1]`. I am also thinking of inputs like `///path//with/too//many/slashes///`

Reply:
I think we should also test and forbid `.` and `..` in the mountpoint path. For example:
```
/./subdir/../../subdir/again///
```
The function should return `-EINVAL` if we see something like this.

Reply:
Ok, I did a quick workaround for having multiple `/`'s in the path. But our `mkdir`/`lookup` implmenetation does not handle such cases, which I find worth fixing sometime.